lib/lightning/pipeline/runner.ex

```diff
@@ -47,21 +48,34 @@ defmodule Lightning.Pipeline.Runner do
         # coveralls-ignore-stop
       end)
 
-      scrubbed_log = Lightning.Scrubber.scrub(scrubber, result.log)
-
       {:ok, run} =
-        update_run(run, %{
-          finished_at: DateTime.utc_now(),
-          exit_code: result.exit_code,
-          log: scrubbed_log
-        })
+        Repo.transaction(fn ->
+          :ok =
+            Lightning.Scrubber.scrub(scrubber, result.log) |> save_run_logs(run)
+
+          {:ok, run} =
+            update_run(run, %{
+              finished_at: DateTime.utc_now(),
+              exit_code: result.exit_code
+            })
+
+          run
+        end)
 
       dataclip_result = Runner.create_dataclip_from_result(result, run)
 
       Lightning.FailureAlerter.alert_on_failure(run)
 
       dataclip_result
     end
+
+    defp save_run_logs(logs, run) do
+      Enum.each(logs, fn log ->
+        if log != "" do
```

Comment:
Interesting... I'm not sure we want to strip out log lines that are empty?

I mean it seems silly to persist them, so maybe the shape of the lines (i.e. whether we include blank ones or not) lives outside of here..

Reply:
Done